sdc/io/csv_ext.py

```diff
@@ -470,43 +593,59 @@ def _gen_csv_reader_py_pyarrow(col_names, col_typs, usecols, sep, typingctx, tar
     return _gen_csv_reader_py_pyarrow_jit_func(csv_reader_py)
 
 
-def _gen_csv_reader_py_pyarrow_func_text_core(col_names, col_typs, usecols, sep, skiprows, signature=None):
+def _gen_csv_reader_py_pyarrow_func_text_core(col_names, col_typs, dtype_present, usecols, signature=None):
     # TODO: support non-numpy types like strings
     date_inds = ", ".join(str(i) for i, t in enumerate(col_typs) if t.dtype == types.NPDatetime('ns'))
-    typ_strs = ", ".join(["{}='{}'".format(to_varname(cname), _get_dtype_str(t))
-                          for cname, t in zip(col_names, col_typs)])
-    pd_dtype_strs = ", ".join(["'{}':{}".format(cname, _get_pd_dtype_str(t)) for cname, t in zip(col_names, col_typs)])
+    return_columns = usecols if usecols and isinstance(usecols[0], str) else col_names
+    nb_objmode_vars = ", ".join([
+        "{}='{}'".format(to_varname(cname), _get_dtype_str(t))
+        for cname, t in zip(return_columns, col_typs)
+    ])
+    pd_dtype_strs = ", ".join([
+        "'{}':{}".format(cname, _get_pd_dtype_str(t))
+        for cname, t in zip(return_columns, col_typs)
+    ])
 
     if signature is None:
-        signature = "fname"
+        signature = "filepath_or_buffer"
     func_text = "def csv_reader_py({}):\n".format(signature)
-    func_text += "  with objmode({}):\n".format(typ_strs)
-    func_text += "    df = pandas_read_csv(fname, names={},\n".format(col_names)
-    func_text += "       parse_dates=[{}],\n".format(date_inds)
-    func_text += "       dtype={{{}}},\n".format(pd_dtype_strs)
-    func_text += "       skiprows={},\n".format(skiprows)
-    func_text += "       usecols={}, sep='{}')\n".format(usecols, sep)
-    for cname in col_names:
+    func_text += "  with objmode({}):\n".format(nb_objmode_vars)
```

Comment:
nit: You can wrap these lines into into list of strings to avoid `func_text +=` copypaste

Reply:
I have tried it. Code is little bit more clear in this implementation even with repetition.